lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2173,38 +2178,37 @@ export class UserAgentApplication {
   // If pCacheStorage is separated from the class object, or passed as a fn param, scopesUtils.ts can be created
 
   /**
-   * Used to validate the scopes input parameter requested  by the developer.
+   * Used to validate the scopes input parameter requested  by the developer and
+   * to remove openid and profile from the list of scopes passed by the developer.
+   * These scopes are added by default.
+   * 
    * @param {Array<string>} scopes - Developer requested permissions. Not all scopes are guaranteed to be included in the access token returned.
    * @ignore
    * @hidden
    */
   // TODO: Check if this can be combined with filterScopes()
-  private validateInputScope(scopes: Array<string>): string {
-    // Check that scopes is not an empty array
-    if (!scopes || scopes.length < 1) {
-      return "Scopes cannot be passed as an empty array";
+  private validateAndFilterInputScope(scopes: Array<string>): Array<string> {
```

Comment:
This function seems responsible for too much maybe... 
It's not a huge deal, but I think you could break up responsibilities and outcomes of this function by having a isValidScopes() function and filterValidScopes() function 
is valid scopes could return a boolean true/false and instead of try catch code you could do this 
```js
if (!isValidScopes(scopes)) {
   throw ClientConfigurationError.createClientIdSingleScopeError(scopes.toString());
}
const filteredScopes = filterValidScopes(scopes)
```